spaces/models.py

```diff
@@ -167,6 +200,18 @@ class Location(models.Model):
         related_name="location",
         on_delete=models.CASCADE,
     )
+    unit = models.OneToOneField(
```

Comment:
Is this some django convention for referencing one to one relations. To me it would be more natural to define these one to one relations the other way around, like adding location to unit instead of adding unit to location. 

Location doesn't really need to know what model references it and the database isn't normalized when doing it this way?

Reply:
I also think it would be more natural to define these in related models, not in Location. They are currently defined in the Location because for some strange reason Django Admin doesn't support one-to-one relations as inline forms without custom templates if relation is defined other way around.  So if you want to be able to fill location in same form as you are creating Unit for example then the relation has to be defined like its defined currently.

If someone has at some point time to investigate a convenient way to show inline location form in django admin and also defining location in the related models (space, unit etc) it would be nice imo too. But changing them now would break the ability to fill location inline in django admin and you would have to create location objects separately and then go to attach them to related models.

Reply:
I agree with you that the Location OneToOne should be defined in the model that needs it. Also having it this way simply touching non-existent location will blow-up. I could try to check this issue.